UnityProject/Assets/Scripts/Systems/Research/Designs.cs

```diff
@@ -4,157 +4,177 @@
-				} else {
-					DesignPass.Description = " *Breathes in through teeth* nahh, no description here mate ";
-				}
+				Globals.InternalIDSearch = new Dictionary<string, Design>();
 
+				string info = Application.dataPath + Designs.Globals.DesignPath;
 
-				if (JsonTechnologies [i].ContainsKey ("Category")) {
-					DesignPass.Category = JsonConvert.DeserializeObject<List<string>> (JsonTechnologies [i] ["Category"].ToString ());
-				} else {
-					DesignPass.Category = new List<string> ();
-				}
+				string[] fileInfo = Directory.GetFiles(info,"*.json");
 
-				if (JsonTechnologies[i].ContainsKey("Materials"))
-				{
-					DesignPass.Materials = JsonConvert.DeserializeObject<Dictionary<string, int>>(JsonTechnologies[i]["Materials"].ToString());
-				}
-				else
+				foreach(string file in fileInfo)
 				{
-					DesignPass.Materials = new Dictionary<string, int>();
-				}
+					string f = file;
 
-				if (JsonTechnologies [i].ContainsKey ("Compatible_machinery")) {
-					DesignPass.CompatibleMachinery = JsonConvert.DeserializeObject<List<string>> (JsonTechnologies [i] ["Compatible_machinery"].ToString ());
-				} else {
-					DesignPass.CompatibleMachinery = new List<string> ();
+					f = f.Replace(Application.dataPath + @"\Resources\", "");
+					f = f.Replace(".json", "");
+
+					Jsons.Add((Resources.Load(f) as TextAsset).ToString());
 				}
 
-				DesignPass.MachineryType = JsonConvert.DeserializeObject<List<string>> (JsonTechnologies [i] ["Machinery_type"].ToString ());
+				new Task(JsonImportInitialization).Start();
+			}
+		}
+		public static class Globals
+		{
+			public static bool IsInitialised = false;
+
+			public static string DesignPath = @"\Resources\Metadata\Designs\";
 
-				if (JsonTechnologies [i].ContainsKey ("Required_reagents")) {
-					DesignPass.RequiredReagents = JsonConvert.DeserializeObject<Dictionary<string,int>> (JsonTechnologies [i] ["Required_reagents"].ToString ());
-				} else {
-					DesignPass.RequiredReagents = new Dictionary<string,int> ();
-				}
+			public static Dictionary<string,Design> InternalIDSearch;
 
-				if (JsonTechnologies [i].ContainsKey ("Make_reagents")) {
-					DesignPass.ReagentsToMake = JsonConvert.DeserializeObject<Dictionary<string,int>> (JsonTechnologies [i] ["Make_reagents"].ToString ());
-				} else {
-					DesignPass.ReagentsToMake = new Dictionary<string,int> ();
+		}
+		private static void JsonImportInitialization()
+		{
+			foreach (string json in Jsons)
+			{
+				var JsonTechnologies = JsonConvert.DeserializeObject<List<Dictionary<String, System.Object>>>(json);
+				for (var i = 0; i < JsonTechnologies.Count(); i++)
+				{
+					Design DesignPass = new Design();
+					DesignPass.InternalID = JsonTechnologies[i]["Internal_ID"].ToString();
+					DesignPass.Name = JsonTechnologies[i]["name"].ToString();
+					DesignPass.ItemID = JsonTechnologies[i]["Item_ID"].ToString();
+
+					if (JsonTechnologies[i].ContainsKey("Description"))
```

Comment:
is it needed to check for every key individually?

Reply:
I think it's best. If a specific item doesn't have a description set for instance- you don't want that throwing an error stopping every other design from importing as well

Reply:
There must be a way to make missing keys a null on the deserialized object

Reply:
The mere act of trying to get the key if it doesnt exist causes an NRE and stops it form imporintg- always have to check with .Contains() first

Reply:
https://docs.microsoft.com/en-us/answers/questions/291110/json-deserialization-behaviour-regarding-missing-p.html

Reply:
json newtonsoft could also use that idk , I think it's okay to have the checking thing

Reply:
Any repetitive thing like that is prone to mistakes and it is also annoying to maintain

Reply:
I tested it in editor- the checks have to be there- the problem isnt the Json getting a null field- the mere act or trying to get an invalid key from the dictionary causes the erorr and breaks it. It doesn't work without the check. (Getting the wrong key doesn't make the dictionary return null- it just makes it cause an error). Not sure what we are doing different to that link you sent- all I know- it has been tested to not work.

Reply:
I would have to see how you did it, you are probably missing giving the model a default value, but I will merge for now